src/lib/hooks/routing/useClientSideSmartOrderRouterTrade.ts

```diff
@@ -0,0 +1,122 @@
+import { Currency, CurrencyAmount, TradeType } from '@uniswap/sdk-core'
+import { useStablecoinAmountFromFiatValue } from 'hooks/useUSDCPrice'
+import { useEffect, useMemo, useState } from 'react'
+import { GetQuoteResult, InterfaceTrade, TradeState } from 'state/routing/types'
+import { computeRoutes, transformRoutesToTrade } from 'state/routing/utils'
+
+import { getClientSideQuote } from './clientSideSmartOrderRouter'
+import { useRoutingAPIArguments } from './useRoutingAPIArguments'
+
+export default function useClientSideSmartOrderRouterTrade<TTradeType extends TradeType>(
+  tradeType: TTradeType,
+  amountSpecified?: CurrencyAmount<Currency>,
+  otherCurrency?: Currency
+): {
+  state: TradeState
+  trade: InterfaceTrade<Currency, Currency, TTradeType> | undefined
+} {
+  const [currencyIn, currencyOut]: [Currency | undefined, Currency | undefined] = useMemo(
+    () =>
+      tradeType === TradeType.EXACT_INPUT
+        ? [amountSpecified?.currency, otherCurrency]
+        : [otherCurrency, amountSpecified?.currency],
+    [amountSpecified, otherCurrency, tradeType]
+  )
+
+  const queryArgs = useRoutingAPIArguments({
+    tokenIn: currencyIn,
+    tokenOut: currencyOut,
+    amount: amountSpecified,
+    tradeType,
+    useClientSideRouter: true,
+  })
+
+  const [loading, setLoading] = useState(false)
+  const [fetchedResult, setFetchedResult] = useState<
+    | {
+        data: GetQuoteResult | undefined
+        error?: unknown
+      }
+    | undefined
+  >()
+
+  useEffect(() => {
+    async function fetchQuote() {
+      if (queryArgs) {
+        try {
+          const result = await getClientSideQuote(queryArgs)
+          setLoading(false)
+          setFetchedResult({
+            data: result.data,
+            error: result.error,
+          })
+        } catch (e) {
+          setLoading(false)
+          setFetchedResult({
+            data: undefined,
+            error: true,
+          })
+        }
+      }
+    }
+    setLoading(true)
```

Comment:
nit: move these calls above the definition and separate with a newline. the definiton will be hoisted, and IMO it's more readable that way.